checkov/kustomize/runner.py

```diff
@@ -118,22 +125,88 @@ def mutate_kubernetes_results(
             record.set_guideline(check.guideline)
             report.add_record(record=record)
 
         return report
 
+    def _get_caller_file_info(self, entity_context: _EntityContext, k8_file: str, k8_file_path: str, resource_id: str,
+                              root_folder: str | None) -> tuple[tuple[int, int] | None, str]:
+        origin_relative_path = entity_context['origin_relative_path']
+        k8s_file_dir = pathlib.Path(k8_file_path).parent
+        raw_file_path = k8s_file_dir / origin_relative_path
+        caller_file_path = self._get_caller_file_path(k8s_file_dir, origin_relative_path, raw_file_path)
+        if root_folder is None:
+            return None, caller_file_path
+        caller_file_line_range = self._get_caller_line_range(root_folder, k8_file, origin_relative_path,
+                                                             resource_id)
+        return caller_file_line_range, caller_file_path
+
+    @staticmethod
+    def _get_caller_file_path(k8s_file_dir: pathlib.Path, origin_relative_path: str, raw_file_path: pathlib.Path)\
+            -> str:
+        amount_of_parents = str.count(origin_relative_path, '..')
+        directory_prefix_path = k8s_file_dir
+        for _ in range(amount_of_parents):
+            directory_prefix_path = directory_prefix_path.parent
+
+        directory_prefix = str(directory_prefix_path)
+        resolved_path = str(raw_file_path.resolve())
+        # Make sure the resolved path starts with the root folder, as pathlib.Path.resolve() might change it
+        if directory_prefix in resolved_path and not resolved_path.startswith(directory_prefix):
+            resolved_path_parts = resolved_path.split(directory_prefix)
+            if len(resolved_path_parts) > 1:
+                resolved_path = f'{directory_prefix}{"".join(resolved_path_parts[1:])}'
+            else:
+                resolved_path = f'{directory_prefix}{"".join(resolved_path_parts)}'
+
+        return resolved_path[len(str(directory_prefix)):]
+
+    def _get_caller_line_range(self, root_folder: str, k8_file: str, origin_relative_path: str,
+                               resource_id: str) -> tuple[int, int] | None:
+        raw_caller_directory = (pathlib.PurePath(k8_file.lstrip(os.path.sep)).parent /
+                                pathlib.PurePath(origin_relative_path.lstrip(os.path.sep)).parent)
+        caller_directory = str(pathlib.Path(f'{os.path.sep}{raw_caller_directory}').resolve())
+        splitted_dir = caller_directory.split(root_folder)
+        if len(splitted_dir) > 1:
+            # Removes any unnecessary additions by `Path.resolve`
+            caller_directory = root_folder + ''.join(splitted_dir[1:])
+        else:
+            caller_directory = root_folder.join(splitted_dir)
+        file_ending = pathlib.Path(origin_relative_path).suffix
+        caller_file_path = f'{str(pathlib.PurePath(caller_directory) / resource_id.replace(".", "-"))}{file_ending}'
```

Comment:
why using `PurePath` here and not just `Path`?

Reply:
in the beginning thought it might help me solve a problem I had but there's no real reason anymore. done